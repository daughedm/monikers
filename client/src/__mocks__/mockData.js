/* eslint-disable max-len */

export const activeCard = {
  name: 'Doge',
  description:
    'An Internet meme that shows a Shiba Inu surrounded by colorful Comic Sans text that describes its inner monologue, such as "Wow," "Concern," and "so scare." There is much confuse over the name\'s pronunciation, yet it was recently used to brand a Bitcoin competitor.',
  category: 'CELEBRITY',
  pointValue: 3
};

export const activeCards = [activeCard];
