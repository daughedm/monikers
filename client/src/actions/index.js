import {
  getTeamNames,
  numOfPlayers,
  currentTeam,
  currentRound,
  teamOneScore,
  teamTwoScore,
  updateTeamTimer,
  addTeamNames,
  resetStore
} from './gameActions/gameActions';

import {
  updateActiveCards,
  discardedCards,
  addCard,
  numOfCards,
  clearDiscardedCards
} from './cardActions/cardActions';

export {
  getTeamNames,
  numOfPlayers,
  addCard,
  discardedCards,
  clearDiscardedCards,
  currentTeam,
  currentRound,
  teamOneScore,
  teamTwoScore,
  updateActiveCards,
  updateTeamTimer,
  numOfCards,
  addTeamNames,
  resetStore
};
