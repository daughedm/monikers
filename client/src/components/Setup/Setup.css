.label-name {
  color: #fff;
  margin: 10px auto 5px auto;
  font-weight: 400;
  font-size: 1rem;
}

.wrapper {
  width: 75%;
  margin: auto;
  padding: 30px 0;
}

.input-field {
  width: 100%;
  margin: auto;
  height: 40px;
  font-size: 1rem;
  color: #555;
  border-radius: 5px;
  border: 1px solid #FFF;
  background-color: #fff;
}

input::placeholder {
  font-size: 1rem;
  text-indent: 10px;
}

input {
  outline: none;
  -webkit-appearance: none;
  text-indent: 10px;
}

.start-button {
  width: 100%;
  height: 40px;
  color: #FFF;
  background-color: rgba(76, 189, 159, 1);
  border: 1px solid #FFF;
  border-radius: 5px;
  font-size: 1rem;
  margin-top: 30px;
  -webkit-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  -moz-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  
}

.back-button {
  width: 100%;
  height: 40px;
  color: #FFF;
  background-color: rgba(134, 106, 173, 1);
  border: 1px solid #FFF;
  border-radius: 5px;
  font-size: 1rem;
  margin-top: 10px;
  -webkit-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  -moz-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  
}

.logo {
  width: 80%;
  margin: 0 auto 20px auto;
  display: block;
}

.game-setup-headline {
  color: #FFF;
  text-align: center;
  margin: 20px;
}

@media (min-width: 360px) {
  .logo {
    margin: 30px auto 30px auto;
  }

  .label-name {
    margin: 20px auto 5px auto;
  }

  .start-button {
    margin-top: 40px;    
  }
}

@media (min-width: 414px) {
  .logo {
    margin: 40px auto 40px auto;
  }

  .input-field {
    height: 45px;
  }
  
  .start-button {
    height: 45px;
  }
  
  .back-button {
    height: 45px;
  }
}

@media (min-width: 620px) {
  .wrapper {
    width: 400px;
    margin: auto;
    padding: 30px 0;
  }


}