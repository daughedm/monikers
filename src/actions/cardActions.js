export const numOfCards = number => ({
  type: 'NUM_OF_CARDS',
  number
});

export const addCard = card => ({
  type: 'ADD_CARD',
  card
});

export const updateActiveCards = cards => ({
  type: 'UPDATE_ACTIVE_CARDS',
  cards
})

export const discardedCards = card => ({
  type: 'DISCARDED_CARDS',
  card
});

export const clearDiscardedCards = card => ({
  type: 'CLEAR_DISCARDED_CARDS',
  card
});
