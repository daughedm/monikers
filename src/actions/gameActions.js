export const getTeamNames = teamName => ({
  type: 'TEAM_NAMES',
  teamName
});

export const numOfPlayers = number => ({
  type: 'NUM_OF_PLAYERS',
  number
});

export const addCard = card => ({
  type: 'ADD_CARD',
  card
});

export const discardedCards = card => ({
  type: 'DISCARDED_CARDS',
  card
});

export const currentTeam = currentTeam => ({
  type: 'CURRENT_TEAM',
  currentTeam
});

export const currentRound = roundNumber => ({
  type: 'CURRENT_ROUND',
  roundNumber
});

export const teamOneScore = points => ({
  type: 'TEAM_ONE_SCORE',
  points
});

export const teamTwoScore = points => ({
  type: 'TEAM_TWO_SCORE',
  points
});
