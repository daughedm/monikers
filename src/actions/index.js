import {
  getTeamNames,
  numOfPlayers
} from './gameActions.js';

export  {
  getTeamNames,
  numOfPlayers
};
