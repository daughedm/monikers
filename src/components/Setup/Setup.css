.label-name {
  color: #fff;
  margin: 20px auto 5px auto;
  font-weight: 400;
  font-size: 1rem;
}

.wrapper {
  width: 75%;
  margin: auto;
  padding: 50px 0;
}

.input-field {
  width: 100%;
  margin: auto;
  height: 40px;
  font-size: 1rem;
  color: #555;
  border-radius: 5px;
  border: 1px solid #FFF;
  background-color: #fff;
}

input::placeholder {
  font-size: 1rem;
  text-indent: 10px;
}

input {
  outline: none;
  -webkit-appearance: none;
  text-indent: 10px;
}

.start-button {
  width: 100%;
  height: 50px;
  color: #FFF;
  background-color: rgba(76, 189, 159, 1);
  border: 1px solid #FFF;
  border-radius: 5px;
  font-size: 1rem;
  margin-top: 40px;
  -webkit-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  -moz-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  
}

.back-button {
  width: 100%;
  height: 50px;
  color: #FFF;
  background-color: rgba(134, 106, 173, 1);
  border: 1px solid #FFF;
  border-radius: 5px;
  font-size: 1rem;
  margin-top: 20px;
  -webkit-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  -moz-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  
}

.logo {
  width: 80%;
  margin: 10px auto 40px auto;
  display: block;
}

.headline {
  color: #FFF;
  text-align: center;
  margin: 50px;
}