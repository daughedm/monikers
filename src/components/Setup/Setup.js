import React, { Component } from 'react';
import PropTypes from 'prop-types';
import indexedDB from '../../indexedBD';
import { connect } from 'react-redux';
import * as actions from '../../actions';
import './Setup.css';
import logo from '../../assets/Monikers_logo_lockup-02.svg';

export class Setup extends Component {
  constructor(props) {
    super(props);

    this.state = {
      teamOne: '',
      teamTwo: '',
      numCards: -1
    };
  }

  componentDidMount() {}

  getActiveCards = async numberOfCards => {
    const { addCard } = this.props;
    const numberOfCardsInIDB = await indexedDB.allCards.count();
    const lowerBound = 1;
    const upperBound = numberOfCardsInIDB;
    const uniqueRandomNumbers = [];

    while (uniqueRandomNumbers.length < numberOfCards) {
      const randomNumber = Math.floor(
        Math.random() * (upperBound - lowerBound) + lowerBound
      );

      if (uniqueRandomNumbers.indexOf(randomNumber) === -1) {
        uniqueRandomNumbers.push(randomNumber);
      }
    }
    const allCards = await indexedDB.allCards.toArray();
    uniqueRandomNumbers.forEach(num => {
      addCard(allCards[num]);
    });
  };

  handleChange = e => {
    const { value, name } = e.target;

    this.setState({
      [name]: value
    });
  };

  handleSubmit = async e => {
    e.preventDefault();
    const { teamOne, teamTwo, numCards } = this.state;

    if (teamOne && teamTwo && numCards >= 30) {
      await this.storeGameInfo(teamOne, teamTwo, numCards);
      this.getActiveCards(numCards);
      this.props.history.push('/play');
    }
  };

  storeGameInfo = async (teamOne, teamTwo, numCards) => {
    const { addTeamNames, numOfCards, currentTeam } = this.props;

    addTeamNames(teamOne);
    addTeamNames(teamTwo);
    numOfCards(parseInt(numCards));
    currentTeam(teamOne);
  };

  handleBackButton = e => {
    e.preventDefault();
    this.props.history.push('/');
  };

  render() {
    return (
      <div className="wrapper">
        <img className="logo" src={logo} alt="Monikers logo" />
        <h2 className="game-setup-headline">Game Setup</h2>
        <form action="" onSubmit={this.handleSubmit}>
          <h3 className="label-name">Team One</h3>
          <input
            className="input-field"
            type="text"
            name="teamOne"
            placeholder="Enter team name"
            onChange={this.handleChange}
          />
          <h3 className="label-name">Team Two</h3>
          <input
            className="input-field"
            type="text"
            name="teamTwo"
            placeholder="Enter team name"
            onChange={this.handleChange}
          />
          <h3 className="label-name">Number of Cards</h3>
          <input
            className="input-field"
            type="number"
            min="30"
            max="60"
            placeholder="Enter number 30-60"
            name="numCards"
            onChange={this.handleChange}
          />
          <button className="start-button" type="submit">
            START GAME
          </button>
          <button
            className="back-button"
            type="submit"
            onClick={this.handleBackButton}
          >
            BACK
          </button>
        </form>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  teamNames: state.teamNames,
  numCards: state.numCards
});

export const mapDispatchToProps = dispatch => ({
  getTeamNames: teamName => dispatch(actions.getTeamNames(teamName)),
  numOfCards: number => dispatch(actions.numOfCards(number)),
  currentTeam: team => dispatch(actions.currentTeam(team)),
  addCard: card => dispatch(actions.addCard(card)),
  addTeamNames: teamName => dispatch(actions.addTeamNames(teamName))
});

Setup.propTypes = {
  teamNames: PropTypes.array.isRequired,
  numCards: PropTypes.number.isRequired,
  getTeamNames: PropTypes.func.isRequired,
  numOfCards: PropTypes.func.isRequired,
  currentTeam: PropTypes.func.isRequired,
  addCard: PropTypes.func.isRequired,
  addTeamNames: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Setup);
