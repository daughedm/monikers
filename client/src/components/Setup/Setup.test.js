/* eslint-disable max-len */

import React from 'react';
import { shallow } from 'enzyme';
import { Setup, mapStateToProps, mapDispatchToProps } from './Setup';
import indexedDB from '../../indexedBD';
jest.mock('../../indexedBD.js');
import * as mockData from '../../__mocks__/mockData';
import * as actions from '../../actions';

describe('setup', () => {
  let setup;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      teamNames: ['team1', 'team2'],
      numCards: 2,
      getTeamNames: jest.fn(),
      numOfCards: jest.fn(),
      currentTeam: jest.fn(),
      addCard: jest.fn(),
      addTeamNames: jest.fn(),
      history: { push: jest.fn() }
    };
    setup = shallow(<Setup {...mockProps} />);
  });

  it('matches the snapshot', () => {
    expect(setup).toMatchSnapshot();
  });

  describe('getActiveCards', () => {
    it.skip('calls addCard', async () => {
      const indexedDB = {
        allCards: {
          count: jest.fn(),
          toArray: jest.fn().mockImplementation(() => mockData.cards)
        }
      };

      setup.instance().getActiveCards(mockProps.numCards);

      expect(mockProps.addCard).toHaveBeenCalledTimes(1);
    });
  });

  describe('handleChange', () => {
    it('sets state with event target name & value', () => {
      const mockEvent = { target: { name: 'teamOne', value: 'Red' } };
      const expected = { teamOne: 'Red' };

      setup.instance().handleChange(mockEvent);

      expect(setup.state()).toEqual(expect.objectContaining(expected));
    });
  });

  describe('handleSubmit', () => {
    it('calls preventDefault', () => {
      const mockEvent = { preventDefault: jest.fn() };

      setup.instance().handleSubmit(mockEvent);

      expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
    });

    it('calls storeGameInfo if teamOne && teamTwo && numCards >= 30', () => {
      setup.setState({ teamOne: 'Red', teamTwo: 'Blue', numCards: 30 });
      const mockEvent = { preventDefault: jest.fn() };
      const storeGameInfo = (setup.instance().storeGameInfo = jest.fn());
      setup.instance().getActiveCards = jest.fn();

      setup.instance().handleSubmit(mockEvent);

      expect(storeGameInfo).toHaveBeenCalledTimes(1);
      expect(storeGameInfo).toHaveBeenCalledWith('Red', 'Blue', 30);
    });

    it('calls getActiveCards if teamOne && teamTwo && numCards >= 30', async () => {
      setup.setState({ teamOne: 'Red', teamTwo: 'Blue', numCards: 30 });
      const mockEvent = { preventDefault: jest.fn() };
      setup.instance().storeGameInfo = jest.fn();
      const getActiveCards = (setup.instance().getActiveCards = jest.fn());

      await setup.instance().handleSubmit(mockEvent);

      expect(getActiveCards).toHaveBeenCalledTimes(1);
      expect(getActiveCards).toHaveBeenCalledWith(30);
    });

    it('calls history.push if teamOne && teamTwo && numCards >= 30', async () => {
      setup.setState({ teamOne: 'Red', teamTwo: 'Blue', numCards: 30 });
      const mockEvent = { preventDefault: jest.fn() };
      setup.instance().storeGameInfo = jest.fn();
      setup.instance().getActiveCards = jest.fn();

      await setup.instance().handleSubmit(mockEvent);

      expect(mockProps.history.push).toHaveBeenCalledTimes(1);
      expect(mockProps.history.push).toHaveBeenCalledWith('/play');
    });
  });

  describe('storeGameInfo', () => {
    it('calls addTeamNames twice', () => {
      const teamOne = 'team1';
      const teamTwo = 'team2';
      const numCards = 30;

      setup.instance().storeGameInfo(teamOne, teamTwo, numCards);

      expect(mockProps.addTeamNames).toHaveBeenCalledTimes(2);
      expect(mockProps.addTeamNames).toHaveBeenCalledWith(teamOne);
      expect(mockProps.addTeamNames).toHaveBeenCalledWith(teamTwo);
    });

    it('calls numOfCards', () => {
      const teamOne = 'team1';
      const teamTwo = 'team2';
      const numCards = 30;

      setup.instance().storeGameInfo(teamOne, teamTwo, numCards);

      expect(mockProps.numOfCards).toHaveBeenCalledTimes(1);
      expect(mockProps.numOfCards).toHaveBeenCalledWith(30);
    });

    it('calls numOfCards', () => {
      const teamOne = 'team1';
      const teamTwo = 'team2';
      const numCards = 30;

      setup.instance().storeGameInfo(teamOne, teamTwo, numCards);

      expect(mockProps.currentTeam).toHaveBeenCalledTimes(1);
      expect(mockProps.currentTeam).toHaveBeenCalledWith(teamOne);
    });
  });

  describe('handleBackButton', () => {
    it('calls preventDefault', () => {
      const mockEvent = { preventDefault: jest.fn() };

      setup.instance().handleBackButton(mockEvent);

      expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
    });

    it('calls history.push', () => {
      const mockEvent = { preventDefault: jest.fn() };

      setup.instance().handleBackButton(mockEvent);

      expect(mockProps.history.push).toHaveBeenCalledTimes(1);
      expect(mockProps.history.push).toHaveBeenCalledWith('/');
    });
  });

  describe('mapStateToProps', () => {
    it('maps state properties to props', () => {
      const mockState = {
        teamNames: actions.teamNames,
        numCards: actions.numCards
      };
      const expected = {
        teamNames: actions.teamNames,
        numCards: actions.numCards
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with the correct params on numOfCards', () => {
      const mockDispatch = jest.fn();
      const number = 30;
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = actions.numOfCards(number);
      mappedProps.numOfCards(number);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('calls dispatch with the correct params on currentTeam', () => {
      const mockDispatch = jest.fn();
      const team = 'team1';
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = actions.currentTeam(team);
      mappedProps.currentTeam(team);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('calls dispatch with the correct params on addCard', () => {
      const mockDispatch = jest.fn();
      const card = {};
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = actions.addCard(card);
      mappedProps.addCard(card);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('calls dispatch with the correct params on addTeamNames', () => {
      const mockDispatch = jest.fn();
      const teamName = 'team1';
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = actions.addTeamNames(teamName);
      mappedProps.addTeamNames(teamName);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
});
