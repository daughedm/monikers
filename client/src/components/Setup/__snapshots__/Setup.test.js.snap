// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setup matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Setup
    addCard={[Function]}
    addTeamNames={[Function]}
    currentTeam={[Function]}
    getTeamNames={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    numCards={2}
    numOfCards={[Function]}
    teamNames={
        Array [
            "team1",
            "team2",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Monikers logo"
          className="logo"
          src="Monikers_logo_lockup-02.svg"
/>,
        <h2
          className="game-setup-headline"
>
          Game Setup
</h2>,
        <form
          action=""
          onSubmit={[Function]}
>
          <h3
                    className="label-name"
          >
                    Team One
          </h3>
          <input
                    autoComplete="off"
                    className="input-field"
                    name="teamOne"
                    onChange={[Function]}
                    placeholder="Enter team name"
                    type="text"
          />
          <h3
                    className="label-name"
          >
                    Team Two
          </h3>
          <input
                    autoComplete="off"
                    className="input-field"
                    name="teamTwo"
                    onChange={[Function]}
                    placeholder="Enter team name"
                    type="text"
          />
          <h3
                    className="label-name"
          >
                    Number of Cards
          </h3>
          <input
                    className="input-field"
                    max="60"
                    min="30"
                    name="numCards"
                    onChange={[Function]}
                    placeholder="Enter number 30-60"
                    type="number"
          />
          <button
                    className="start-button"
                    type="submit"
          >
                    START GAME
          </button>
          <button
                    className="back-button"
                    onClick={[Function]}
                    type="submit"
          >
                    BACK
          </button>
</form>,
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Monikers logo",
          "className": "logo",
          "src": "Monikers_logo_lockup-02.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Game Setup",
          "className": "game-setup-headline",
        },
        "ref": null,
        "rendered": "Game Setup",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "",
          "children": Array [
            <h3
              className="label-name"
>
              Team One
</h3>,
            <input
              autoComplete="off"
              className="input-field"
              name="teamOne"
              onChange={[Function]}
              placeholder="Enter team name"
              type="text"
/>,
            <h3
              className="label-name"
>
              Team Two
</h3>,
            <input
              autoComplete="off"
              className="input-field"
              name="teamTwo"
              onChange={[Function]}
              placeholder="Enter team name"
              type="text"
/>,
            <h3
              className="label-name"
>
              Number of Cards
</h3>,
            <input
              className="input-field"
              max="60"
              min="30"
              name="numCards"
              onChange={[Function]}
              placeholder="Enter number 30-60"
              type="number"
/>,
            <button
              className="start-button"
              type="submit"
>
              START GAME
</button>,
            <button
              className="back-button"
              onClick={[Function]}
              type="submit"
>
              BACK
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Team One",
              "className": "label-name",
            },
            "ref": null,
            "rendered": "Team One",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "off",
              "className": "input-field",
              "name": "teamOne",
              "onChange": [Function],
              "placeholder": "Enter team name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Team Two",
              "className": "label-name",
            },
            "ref": null,
            "rendered": "Team Two",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "off",
              "className": "input-field",
              "name": "teamTwo",
              "onChange": [Function],
              "placeholder": "Enter team name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Number of Cards",
              "className": "label-name",
            },
            "ref": null,
            "rendered": "Number of Cards",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "max": "60",
              "min": "30",
              "name": "numCards",
              "onChange": [Function],
              "placeholder": "Enter number 30-60",
              "type": "number",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "START GAME",
              "className": "start-button",
              "type": "submit",
            },
            "ref": null,
            "rendered": "START GAME",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BACK",
              "className": "back-button",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "BACK",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Monikers logo"
            className="logo"
            src="Monikers_logo_lockup-02.svg"
/>,
          <h2
            className="game-setup-headline"
>
            Game Setup
</h2>,
          <form
            action=""
            onSubmit={[Function]}
>
            <h3
                        className="label-name"
            >
                        Team One
            </h3>
            <input
                        autoComplete="off"
                        className="input-field"
                        name="teamOne"
                        onChange={[Function]}
                        placeholder="Enter team name"
                        type="text"
            />
            <h3
                        className="label-name"
            >
                        Team Two
            </h3>
            <input
                        autoComplete="off"
                        className="input-field"
                        name="teamTwo"
                        onChange={[Function]}
                        placeholder="Enter team name"
                        type="text"
            />
            <h3
                        className="label-name"
            >
                        Number of Cards
            </h3>
            <input
                        className="input-field"
                        max="60"
                        min="30"
                        name="numCards"
                        onChange={[Function]}
                        placeholder="Enter number 30-60"
                        type="number"
            />
            <button
                        className="start-button"
                        type="submit"
            >
                        START GAME
            </button>
            <button
                        className="back-button"
                        onClick={[Function]}
                        type="submit"
            >
                        BACK
            </button>
</form>,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Monikers logo",
            "className": "logo",
            "src": "Monikers_logo_lockup-02.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Game Setup",
            "className": "game-setup-headline",
          },
          "ref": null,
          "rendered": "Game Setup",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "action": "",
            "children": Array [
              <h3
                className="label-name"
>
                Team One
</h3>,
              <input
                autoComplete="off"
                className="input-field"
                name="teamOne"
                onChange={[Function]}
                placeholder="Enter team name"
                type="text"
/>,
              <h3
                className="label-name"
>
                Team Two
</h3>,
              <input
                autoComplete="off"
                className="input-field"
                name="teamTwo"
                onChange={[Function]}
                placeholder="Enter team name"
                type="text"
/>,
              <h3
                className="label-name"
>
                Number of Cards
</h3>,
              <input
                className="input-field"
                max="60"
                min="30"
                name="numCards"
                onChange={[Function]}
                placeholder="Enter number 30-60"
                type="number"
/>,
              <button
                className="start-button"
                type="submit"
>
                START GAME
</button>,
              <button
                className="back-button"
                onClick={[Function]}
                type="submit"
>
                BACK
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Team One",
                "className": "label-name",
              },
              "ref": null,
              "rendered": "Team One",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "className": "input-field",
                "name": "teamOne",
                "onChange": [Function],
                "placeholder": "Enter team name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Team Two",
                "className": "label-name",
              },
              "ref": null,
              "rendered": "Team Two",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "className": "input-field",
                "name": "teamTwo",
                "onChange": [Function],
                "placeholder": "Enter team name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Number of Cards",
                "className": "label-name",
              },
              "ref": null,
              "rendered": "Number of Cards",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "max": "60",
                "min": "30",
                "name": "numCards",
                "onChange": [Function],
                "placeholder": "Enter number 30-60",
                "type": "number",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "START GAME",
                "className": "start-button",
                "type": "submit",
              },
              "ref": null,
              "rendered": "START GAME",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "BACK",
                "className": "back-button",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "BACK",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
